### PR TITLE
Update column filtering

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: miso-web/src/main/webapp/scripts/sample_hot.js [hunk 1]
```diff
@@ -587,338 +587,290 @@ Sample.hot = {
...
+        include: detailedBool && idColBoolean
+      },
+      {
+        header: 'Tissue Class',
+        data: 'parentSampleClassAlias',
+        type: 'dropdown',
+        trimDropdown: false,
+        source: Sample.hot.getTissueClassesAliasOnly(),
+        validator: validateTissueClasses,
+        include: detailedBool && idColBoolean && (sampleCategory === 'Aliquot' || sampleCategory === 'Stock')
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/sample_hot.js]: Shouldn't be necessary on aliquots

**Comment by callunity** [comment]
:+1: logic looks good

**Comment by apmasell** [comment]
Okay. This is ready for real review. ^_^

File: miso-web/src/main/webapp/scripts/sample_hot.js [hunk 2]
```diff
@@ -640,45 +656,87 @@ Sample.hot = {
...
       {
-        header: 'External Name',
-        data: 'externalName',
-        type: 'text',
-        validator: requiredText,
-        include: detailedBool && idColBoolean
-      },{
-        header: 'Sex',
-        data: 'donorSex',
+        header: 'Kit',
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/sample_hot.js]: Can you confirm that it's the Aliquots which have the kits applied to them, and not the Stocks? I would have thought that an extraction kit (chemical change) happens at the stock level, not the aliquot level.

File: miso-web/src/main/webapp/scripts/sample_hot.js [hunk 3]
```diff
@@ -578,347 +563,339 @@ Sample.hot = {
...
+        type: 'numeric',
+        format: '0.00',
+        include: showQcs || show['Stock']
+      },
+      {
+        header: 'Conc.',
+        data: 'concentration',
+        type: 'numeric',
+        format: '0.00',
+        include: showQcs || show['Stock']
```
Comment (djcooke) [inline miso-web/src/main/webapp/scripts/sample_hot.js]: Concentration is on stock only

**Comment by djcooke** [comment]
When creating a Tissue, both 'Sample Class' and 'Tissue Class' columns are shown, and required. Setting them both results in:

`Row 1: Parent Primary Tumor Tissue not permitted to have a child of type Primary Tumor Tissue`

The Tissue Class column was meant to be shown only when receiving Stocks in order to determine the parent (ghost) tissue's SampleClass. TissueClass should be null if SampleClass is a tissue class

**Comment by djcooke** [comment]
When propagating tissue > stock, STR Status 'Not Submitted' should be selected by default. Not sure if this was broken here or during merge of sample model refactor

**Comment by apmasell** [comment]
No, it was working after the merge.

**Comment by apmasell** [comment]
Okay. It wasn't working after the merge, because it never worked for propagate.

**Comment by djcooke** [comment]
Ah, a test case I missed in my own fix then - my bad!

:+1:

**Pull request merged**